src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -8,288 +11,100 @@ import {
 } from '../utils/performance-utils';
 
 export const NAVIGATION_EVENT_PLUGIN_ID = 'navigation';
-
 const NAVIGATION = 'navigation';
-const LOAD = 'load';
 
-/**
- * This plugin records performance timing events generated during every page load/re-load activity.
- * Paint, resource and performance event types make sense only if all or none are included.
- * For RUM, these event types are inter-dependent. So they are recorded under one plugin.
- */
+/** This plugin records performance timing events generated during every page load/re-load activity. */
 export class NavigationPlugin extends InternalPlugin {
     private config: PerformancePluginConfig;
+    private po: PerformanceObserver;
     constructor(config?: PartialPerformancePluginConfig) {
         super(NAVIGATION_EVENT_PLUGIN_ID);
         this.config = { ...defaultPerformancePluginConfig, ...config };
+        this.po = new PerformanceObserver(this.performanceEntryHandler);
     }
 
     enable(): void {
         if (this.enabled) {
             return;
         }
         this.enabled = true;
-        window.addEventListener(LOAD, this.eventListener);
+        this.onload();
     }
 
     disable(): void {
         if (!this.enabled) {
             return;
         }
         this.enabled = false;
-        if (this.eventListener) {
-            window.removeEventListener(LOAD, this.eventListener);
-        }
-    }
-
-    /**
-     * Use the loadEventEnd field from window.performance to check if the website
-     * has loaded already.
-     *
-     * @returns boolean
-     */
-    hasTheWindowLoadEventFired() {
-        if (
-            window.performance &&
-            window.performance.getEntriesByType(NAVIGATION).length
-        ) {
-            const navData = window.performance.getEntriesByType(
-                NAVIGATION
-            )[0] as PerformanceNavigationTiming;
-            return Boolean(navData.loadEventEnd);
-        }
-        return false;
+        this.po.disconnect();
     }
 
     /**
-     * Use Navigation timing Level 1 for all browsers by default -
-     * https://developer.mozilla.org/en-US/docs/Web/API/Performance/timing
-     *
-     * If browser provides support, use Navigation Timing Level 2 specification -
+     * Callback to record PerformanceNavigationTiming as RUM NavigationEvent
      * https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming
-     *
-     * Only the current document resource is included in the performance timeline;
-     * there is only one PerformanceNavigationTiming object in the performance timeline.
-     * https://www.w3.org/TR/navigation-timing-2/
      */
-    eventListener = () => {
-        if (performance.getEntriesByType(NAVIGATION).length === 0) {
-            this.performanceNavigationEventHandlerTimingLevel1();
```

Comment:
Remove support for Level1 timing, since browser compatibility for Level2 is very strong

https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming#browser_compatibility

The only previous concern was Safari version older than v15, but customer use is extremely low for those versions.

https://www.w3schools.com/browsers/browsers_safari.asp

Reply:
What happens when the user has a browser that doesn't support PerformanceNavigationTiming?

Reply:
If navigation is not supported, then our performance observer will simply never detect and PerformanceNavigationTiming entries. I have added a runtime check for support so that if the browser is incompatible, then we do not initiate an observer.